gRPC/Client Examples/StimTherapyApp/MainWindow.xaml.cs

```diff
@@ -153,66 +155,96 @@ private void MainWindow_Loaded(object sender, RoutedEventArgs e)
                     // disable buttons
                     btn_beta.IsEnabled = false; // beta stim button; have to use method invoker
                     btn_open.IsEnabled = false; // open loop stim button
-                    btn_diagnostic.IsEnabled = false; // diagnostics button
                     btn_stop.IsEnabled = false; // stop stim button
 
-                    // temporary- hide diagnostic buttons
-                    btn_diagnostic.Visibility = Visibility.Hidden;
                 }));
 
             // Start update timer
             neuroChartUpdateTimer = new System.Timers.Timer(200);
             neuroChartUpdateTimer.Elapsed += neuroChartUpdateTimer_Elapsed;
             neuroChartUpdateTimer.Start();
-        }
 
-        private void neuroChartUpdateTimer_Elapsed(object sender, ElapsedEventArgs e)
+            connectionUpdateTimer = new System.Timers.Timer(200);
+            connectionUpdateTimer.Elapsed += connectionUpdateTimer_Elapsed;
+            connectionUpdateTimer.Start();
+        }
+        
+        private void connectionUpdateTimer_Elapsed(object sender, ElapsedEventArgs e)
```

Comment:
it seems like we should be able to have the connection async monitor do the bicmanager.dispose itself, and then this could simply check if the bicmanager is disposed so then we don't have to have this whole buffer system? It seems funny to have the main window be the one to monitor the result of the connection stream result